src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy

```diff
@@ -71,55 +68,42 @@ class AssertJGeneratorGradlePlugin implements Plugin<Project> {
 
         def javaPlugin = project.getConvention().getPlugin(JavaPluginConvention)
         // So now we have to go through and add the properties that we want
-        javaPlugin.sourceSets.all { sourceSet ->
+        javaPlugin.sourceSets.all { SourceSet sourceSet ->
             // For each sourceSet we're enacting an action on each one that adds an assertJ generation task to it
             logger.info("sourceSet: ${sourceSet} creating tasks")
 
-            // Get the convention and add the properties
-            Convention sourceSetConvention = new DslObject(sourceSet).convention
-
-            // Create the assertJ closure within the source set, e.g. main { assertJ { } }
-            DefaultAssertJGeneratorSourceSet assertJSourceSet = new DefaultAssertJGeneratorSourceSet(
-                    sourceSet, sourceDirectorySetFactory)
-            sourceSetConvention.plugins[AssertJGeneratorSourceSet.NAME] = assertJSourceSet
-            sourceSet.allSource.source(assertJSourceSet.assertJ)
+            def options = sourceSet.extensions.create(
+                    "assertJ",
+                    DefaultAssertJGeneratorOptions,
+                    objects,
+                    project,
+                    sourceSet,
+            )
 
-            addAndConfigureAssertJGenerate(project, javaPlugin, sourceSet, assertJSourceSet)
+            addAndConfigureAssertJGenerate(project, javaPlugin, sourceSet, options)
         }
     }
 
     // Configures the "generate*" tasks to generate files
-    private static void addAndConfigureAssertJGenerate(final Project project,
-                                                       final JavaPluginConvention javaPlugin,
-                                                       final SourceSet sourceSet,
-                                                       final AssertJGeneratorSourceSet assertJSS) {
+    private void addAndConfigureAssertJGenerate(final Project project,
+                                                final JavaPluginConvention javaPlugin,
+                                                final SourceSet sourceSet,
+                                                final AssertJGeneratorOptions options) {
         // Use the name via calling sourceSet#getTaskName(String, String)
         String generateTaskName = sourceSet.getTaskName('generate', 'assertJ')
 
         logger.info("generationTask: ${generateTaskName}, sourceSet: ${sourceSet}")
 
         // Create a new task for the source set
-        AssertJGenerationTask generationTask = project.tasks.findByName(generateTaskName) as AssertJGenerationTask
-
-        if (!generationTask) {
-            generationTask = project.tasks.create(generateTaskName, AssertJGenerationTask) {
-                description = "Generates AssertJ assertions for the ${sourceSet} sources."
-                generationClasspath = sourceSet.runtimeClasspath // Get the classes used when creating the ClassLoader for
```

Comment:
All of this `init` code moved into the constructor for the `AssertJGenerationTask`.